### ISSUE TITLE
Chef14: yum_package resource fails with runtime error-  yum-helper.py had stderr/stdout output: 

### ISSUE THREAD

**Issue by bipinbachhao** [chef/chef#7283]
<!--- 
!!!!!! NOTE: CHEF CLIENT BUGS ONLY !!!!!!

This issue tracker is for the code contained within this repo -- `chef-client`, base `knife` functionality (not
plugins), `chef-apply`, `chef-solo`, `chef-client -z`, etc.

* Requests for new or alternative functionality should be made to [feedback.chef.io](https://feedback.chef.io/forums/301644-chef-product-feedback/category/110832-chef-client)
* [Chef Server issues](https://github.com/chef/chef-server/issues/new)
* [ChefDK issues](https://github.com/chef/chef-dk/issues/new)
* Cookbook Issues (see the https://github.com/chef-cookbooks repos or search [Supermarket](https://supermarket.chef.io) or GitHub/Google)

-->

## Description
<!--- Briefly describe the issue -->
yum_package failing in RHEL 7.4 Red Hat server, subscribed to Satellite for rpm repos, same cookbook with fine if chef-client downgraded to chef 13.8.5 or 14.0.24
## Chef Version
<!--- Tell us which version of chef-client you are using (see below for Server+ChefDK bugs). -->
14.0.202
## Platform Version
<!--- Tell us which Operating System distribution and version chef-client is running on. -->
RHEL 7.4
## Replication Case
<!--- Tell us what steps to take to replicate your problem.  See [How to create a Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve)
for information on how to create a good replication case. -->
Create new RHEL 7.4 instance with chef-client 14.0.202 hooked into Red Hat Satellite. Run build essential cookbook
## Client Output
<!--- The relevant output of the chef-client run or a link to a gist of the entire run, if there is one.

The debug output (chef-client -l debug) may be useful, but please link to a gist, or truncate it. -->

```
Recipe: build-essential::default                                                                                                                           
  * build_essential[install_packages] action install[2018-05-17T15:39:14-04:00] INFO: Processing build_essential[install_packages] action install (build-essential::default line 22)                                                                                                                                  

    * yum_package[autoconf, bison, flex, gcc, gcc-c++, gettext, kernel-devel, make, m4, ncurses-devel, patch] action install[2018-05-17T15:39:14-04:00] INFO: Processing yum_package[autoconf, bison, flex, gcc, gcc-c++, gettext, kernel-devel, make, m4, ncurses-devel, patch] action install (/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/resource/build_essential.rb line 40)                                                                         

      
      ================================================================================
      Error executing action `install` on resource 'yum_package[autoconf, bison, flex, gcc, gcc-c++, gettext, kernel-devel, make, m4, ncurses-devel, patch]'                                                                                                                                                          
      ================================================================================                                                                     
                                                                                                                                                           
      RuntimeError                                                                                                                                         
      ------------                                                                                                                                         
      yum-helper.py had stderr/stdout output:                                                                                                              
                                                                                                                                                           
      BDB2053 Freeing read locks for locker 0x40e: 3464/139965932410688                                                                                    
      BDB2053 Freeing read locks for locker 0x410: 3464/139965932410688                                                                                    
      Traceback (most recent call last):                                                                                                                   
        File "/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/provider/package/yum/yum_helper.py", line 197, in <module>                
          query(command)                                                                                                                                   
        File "/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/provider/package/yum/yum_helper.py", line 147, in query                   
          pkgs = base.bestPackagesFromList(pkgs,single_name=True)                                                                                          
        File "/usr/lib/python2.7/site-packages/yum/__init__.py", line 4441, in bestPackagesFromList                                                        
          multi = self._bestPackageFromList(multiLib, req=req)                                                                                             
        File "/usr/lib/python2.7/site-packages/yum/__init__.py", line 4408, in _bestPackageFromList                                                        
          bestlist = self._compare_providers(pkglist, reqpo=None, req=req)                                                                                 
        File "/usr/lib/python2.7/site-packages/yum/depsolve.py", line 1590, in _compare_providers                                                          
          pkgresults[po] += (100 - po.repo.compare_providers_priority) * 10                                                                                
      AttributeError: FakeRepository instance has no attribute 'compare_providers_priority'                                                                
      Loaded plugins: langpacks, ovl, product-id, subscription-manager, versionlock                                                                        
                                                                                                                                                           
      Resource Declaration:                                                                                                                                
      ---------------------                                                                                                                                
      # In /opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/resource/build_essential.rb                                                  
                                                                                                                                                           
       40:           declare_resource(:package,  %w{ autoconf bison flex gcc gcc-c++ gettext kernel-devel make m4 ncurses-devel patch })                   
       41:                                                                                                                                                 
                                                                                                                                                           
      Compiled Resource:                                                                                                                                   
      ------------------                                                                                                                                   
      # Declared in /opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/resource/build_essential.rb:40:in `block in <class:BuildEssential>' 
                                                                                                                                                           
      yum_package("autoconf, bison, flex, gcc, gcc-c++, gettext, kernel-devel, make, m4, ncurses-devel, patch") do                                         
        package_name ["autoconf", "bison", "flex", "gcc", "gcc-c++", "gettext", "kernel-devel", "make", "m4", "ncurses-devel", "patch"]                    
        action [:install]                                                                                                                                  
        default_guard_interpreter :default                                                                                                                 
        declared_type :package                                                                                                                             
        cookbook_name "build-essential"                                                                                                                    
      end                                                                                                                                                  
                                                                                                                                                           
      System Info:                                                                                                                                         
      ------------                                                                                                                                         
      chef_version=14.0.202                                                                                                                                
      platform=redhat                                                                                                                                      
      platform_version=7.4                                                                                                                                 
      ruby=ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]                                                                                        
      program_name=/usr/bin/chef-client                                                                                                                    
      executable=/opt/chef/bin/chef-client                                                                                                                 
                                                                                                                                                           
[2018-05-17T15:39:21-04:00] INFO: Running queued delayed notifications before re-raising exception                                                         
                                                                                                                                                           
    ================================================================================                                                                       
    Error executing action `install` on resource 'build_essential[install_packages]'                                                                       
    ================================================================================                                                                       
                                                                                                                                                           
    RuntimeError                                                                                                                                           
    ------------                                                                                                                                           
    yum_package[autoconf, bison, flex, gcc, gcc-c++, gettext, kernel-devel, make, m4, ncurses-devel, patch] (/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/resource/build_essential.rb line 40) had an error: RuntimeError: yum-helper.py had stderr/stdout output:                              
                                                                                                                                                           
    BDB2053 Freeing read locks for locker 0x40e: 3464/139965932410688                                                                                      
    BDB2053 Freeing read locks for locker 0x410: 3464/139965932410688                                                                                      
    Traceback (most recent call last):                                                                                                                     
      File "/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/provider/package/yum/yum_helper.py", line 197, in <module>                  
        query(command)                                                                                                                                     
      File "/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/provider/package/yum/yum_helper.py", line 147, in query                     
        pkgs = base.bestPackagesFromList(pkgs,single_name=True)                                                                                            
      File "/usr/lib/python2.7/site-packages/yum/__init__.py", line 4441, in bestPackagesFromList                                                          
        multi = self._bestPackageFromList(multiLib, req=req)                                                                                               
      File "/usr/lib/python2.7/site-packages/yum/__init__.py", line 4408, in _bestPackageFromList                                                          
        bestlist = self._compare_providers(pkglist, reqpo=None, req=req)                                                                                   
      File "/usr/lib/python2.7/site-packages/yum/depsolve.py", line 1590, in _compare_providers                                                            
        pkgresults[po] += (100 - po.repo.compare_providers_priority) * 10                                                                                  
    AttributeError: FakeRepository instance has no attribute 'compare_providers_priority'                                                                  
    Loaded plugins: langpacks, ovl, product-id, subscription-manager, versionlock                                                                          
                                                                                                                                                           
    Resource Declaration:                                                                                                                                  
    ---------------------                                                                                                                                  
    # In /opt/chef/cache/cookbooks/build-essential/recipes/default.rb                                                                                      
                                                                                                                                                           
     22: build_essential 'install_packages' do                                                                                                             
     23:   compile_time node['build-essential']['compile_time']                                                                                            
     24: end                                                                                                                                               
                                                                                                                                                           
    Compiled Resource:                                                                                                                                     
    ------------------                                                                                                                                     
    # Declared in /opt/chef/cache/cookbooks/build-essential/recipes/default.rb:22:in `from_file'                                                           
                                                                                                                                                           
    build_essential("install_packages") do                                                                                                                 
      action [:install]                                                                                                                                    
      default_guard_interpreter :default                                                                                                                   
      declared_type :build_essential                                                                                                                       
      cookbook_name "build-essential"                                                                                                                      
      recipe_name "default"                                                                                                                                
      compile_time false                                                                                                                                   
    end                                                                                                                                                    
                                                                                                                                                           
    System Info:
    ------------
    chef_version=14.0.202
    platform=redhat
    platform_version=7.4
    ruby=ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
    program_name=/usr/bin/chef-client
    executable=/opt/chef/bin/chef-client

[2018-05-17T15:39:21-04:00] INFO: Running queued delayed notifications before re-raising exception

Running handlers:
[2018-05-17T15:39:21-04:00] ERROR: Running exception handlers
Running handlers complete
[2018-05-17T15:39:21-04:00] ERROR: Exception handlers complete
Chef Client failed. 1 resources updated in 25 seconds
[2018-05-17T15:39:21-04:00] INFO: Sending resource update report (run-id: 78e19c67-2d2d-4751-bba8-76516e165be7)
[2018-05-17T15:39:21-04:00] FATAL: Stacktrace dumped to /opt/chef/cache/chef-stacktrace.out
[2018-05-17T15:39:21-04:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2018-05-17T15:39:21-04:00] FATAL: RuntimeError: build_essential[install_packages] (build-essential::default line 22) had an error: RuntimeError: yum_package[autoconf, bison, flex, gcc, gcc-c++, gettext, kernel-devel, make, m4, ncurses-devel, patch] (/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/resource/build_essential.rb line 40) had an error: RuntimeError: yum-helper.py had stderr/stdout output:

BDB2053 Freeing read locks for locker 0x40e: 3464/139965932410688
BDB2053 Freeing read locks for locker 0x410: 3464/139965932410688
Traceback (most recent call last):
  File "/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/provider/package/yum/yum_helper.py", line 197, in <module>
    query(command)
  File "/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.0.202/lib/chef/provider/package/yum/yum_helper.py", line 147, in query
    pkgs = base.bestPackagesFromList(pkgs,single_name=True)
  File "/usr/lib/python2.7/site-packages/yum/__init__.py", line 4441, in bestPackagesFromList
    multi = self._bestPackageFromList(multiLib, req=req)
  File "/usr/lib/python2.7/site-packages/yum/__init__.py", line 4408, in _bestPackageFromList
    bestlist = self._compare_providers(pkglist, reqpo=None, req=req)
  File "/usr/lib/python2.7/site-packages/yum/depsolve.py", line 1590, in _compare_providers
    pkgresults[po] += (100 - po.repo.compare_providers_priority) * 10
AttributeError: FakeRepository instance has no attribute 'compare_providers_priority'
Loaded plugins: langpacks, ovl, product-id, subscription-manager, versionlock
```

## Stacktrace
<!--- Please include the stacktrace.out output or link to a gist of it, if there is one. -->
See Above.

**Comment by thommay** [comment]
We've made a number of fixes to the yum_package provider since 14.0.202; please can you test with the latest stable release?

**Comment by bipinbachhao** [comment]
As @thommay suggested i tried latest stable release Chef: 14.1.12 and issue does not occur
So this issue seems resolved in latest stable release.
